pkg/clients/s3/bucket.go

```diff
@@ -211,3 +212,42 @@ func UpdateBucketACL(ctx context.Context, client BucketClient, bucket *v1beta1.B
 	_, err := client.PutBucketAclRequest(config).Send(ctx)
 	return err
 }
+
+// CopyTag converts a local v1beta.Tag to an S3 Tag
+func CopyTag(tag *v1beta1.Tag) *s3.Tag {
+	if tag == nil {
+		return nil
+	}
+	return &s3.Tag{
+		Key:   aws.String(tag.Key),
+		Value: aws.String(tag.Value),
+	}
+}
+
+// CopyTags converts a list of local v1beta.Tags to S3 Tags
+func CopyTags(tags []v1beta1.Tag) []s3.Tag {
+	if tags == nil {
+		return nil
+	}
+	out := make([]s3.Tag, len(tags))
+	for i := range tags {
+		out[i] = *CopyTag(&tags[i])
+	}
+	return out
+}
+
+// SortS3TagSet stable sorts an external s3 tag list by the key and value.
+func SortS3TagSet(tags []s3.Tag) []s3.Tag {
+	if len(tags) == 0 {
+		return tags
+	}
+	sort.SliceStable(tags, func(i, j int) bool {
+		if aws.StringValue(tags[i].Key) < aws.StringValue(tags[j].Key) {
+			return true
+		} else if aws.StringValue(tags[i].Key) == aws.StringValue(tags[j].Key) {
+			return aws.StringValue(tags[i].Value) < aws.StringValue(tags[j].Value)
```

Comment:
for `ExampleSliceStable`, the comment states "Sort by name, preserving original order", however, in this scenario we can't really trust the sorting from AWS at all, since I couldn't find any documentation that clearly indicated how it is ordered. So instead, I opted to sort by both since we can handle duplicate keys inside of a list of tags.

Reply:
Oh sorry, I thought the line compared keys but it seems it compares values. What I meant was this line should handle it:
```
return aws.StringValue(tags[i].Key) < aws.StringValue(tags[j].Key)
```

I think it's fair to expect that keys are unique in the array.

Reply:
https://docs.aws.amazon.com/general/latest/gr/aws_tagging.html - nvm! did some more research, keys do **need** to be unique inside the array.